pagopa/io-app
# fix: [IOBP-98] Add barcode scan result feedback

## Short description
This PR adds a visual and haptic feedback for the barcode scan result.
- **Successfull scan**: haptic feedback `notificationSuccess`
- **Unsuccessfull scan**: Error toast with haptic feedback `notificationError`

## List of changes proposed in this pull request
- Added `barcodeTypes` props to `useIOBarcodeScanner` and `useIOBarcodeFileReader`, which let developers to enable/disable certain barcode types in different screens/components.
- Added error toast and haptic feedback on barcode scan error handlers
- Added haptic feedback on barcode scan success handlers

## How to test
To test this PR is preferred to have a real device. Emulators could not reproduce haptic feedbacks.
Using the `io-dev-server`, try to scan an IDPay QRCode and check that there is the expected behavior.
